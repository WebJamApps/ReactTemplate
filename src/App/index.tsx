import {
  BrowserRouter, Routes, Route, Navigate,
} from 'react-router-dom';
import { Page1 } from '../containers/Page1';
import { Page2 } from '../containers/Page2';
import { AppTemplate } from './AppTemplate';

export function App(): JSX.Element {
  return (
    <div id="App" className="App">
      <BrowserRouter>
        <AppTemplate>
          <Routes>
            <Route path="/" element={<Page1 />} />
            <Route path="/page2" element={<Page2 />} />
            <Route path="*" element={<Navigate to="/" replace />} />
          </Routes>
        </AppTemplate>
      </BrowserRouter>
    </div>
  );
}
