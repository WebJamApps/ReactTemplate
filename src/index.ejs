<!DOCTYPE html>
<html>

<head prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb#">
  <meta property="og:title" content="React Template" />
  <meta property="og:description" content="A react template project">
  <meta property="og:image"
    content="https://miro.medium.com/v2/resize:fit:720/format:webp/1*y6C4nSvy2Woe0m7bWEn4BA.png">
  <meta property="og:url" content="https://www.react.com">
  <title>
    <%- htmlWebpackPlugin.options.metadata.title %>
  </title>
  <base href="<%- htmlWebpackPlugin.options.metadata.baseUrl %>" />
  <link href="https://fonts.googleapis.com/css?family=PT Sans Caption" rel="stylesheet" />
  <link href="https://fonts.googleapis.com/css?family=Habibi" rel="stylesheet" />
  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet" />
  <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.8.1/css/all.css"
    integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf" crossorigin="anonymous" />

  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <!-- Begin Constant Contact Active Forms -->
  <script> var _ctct_m = "01cd950f6ed99253e212302d6c939739"; </script>
  <script id="signupScript" src="//static.ctctcdn.com/js/signup-form-widget/current/signup-form-widget.min.js" async
    defer></script>
  <!-- End Constant Contact Active Forms -->
</head>

<body>
  <div>
    <div id="root">
      <div class="splash">
        <div class="message">
          <%- htmlWebpackPlugin.options.metadata.title %>
        </div>
        <i class="fa fa-spinner fa-spin"></i>
      </div>
    </div>
  </div>
</body>

</html>
