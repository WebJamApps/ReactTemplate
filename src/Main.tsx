import { StrictMode } from 'react';
import {
  ApolloClient, InMemoryCache, ApolloProvider,
} from '@apollo/client';
import { ReactNotifications } from 'react-notifications-component';
import { createRoot } from 'react-dom/client';
import { App } from './App';
import { AuthProvider } from './providers/Auth.provider';
import '../static/styles.scss';
import { ContentProvider } from './providers/Content.provider';

export const client = new ApolloClient({
  uri: `${process.env.BACKEND_URL}/graphql`, // TODO use same backendurl here
  cache: new InMemoryCache(),
});

const root = createRoot(document.getElementById('root') as HTMLElement);
function Main() {
  return (
    <ApolloProvider client={client}>
      <AuthProvider>
        <ContentProvider>
          <ReactNotifications />
          <App />
        </ContentProvider>
      </AuthProvider>
    </ApolloProvider>
  );
}
const renderMain = (): void => {
  root.render(
    <StrictMode>
      <Main />
    </StrictMode>,
  );
};

renderMain();
