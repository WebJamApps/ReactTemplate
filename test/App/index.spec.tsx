import renderer from 'react-test-renderer';
import { App } from 'src/App';
import commonUtils from 'src/lib/commonUtils';

describe('App component', () => {
  it('is defined', () => {
    expect(App).toBeDefined();
  });
  it('renders the component', () => {
    commonUtils.setTitleAndScroll = jest.fn();
    const href = 'http://localhost:7777';
    const reload = jest.fn();
    Object.defineProperty(window, 'location', {
      value: {
        href, assign: () => { }, reload, origin: href,
      },
      writable: true,
    });
    const app = renderer.create(<App />).toJSON();
    expect(app).toMatchSnapshot();
  });
});
