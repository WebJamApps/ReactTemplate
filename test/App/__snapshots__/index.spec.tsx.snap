// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders the component 1`] = `
<div
  className="App"
  id="App"
>
  <div
    className="page-host"
  >
    <div
      className="home-sidebar close drawer-container"
      id="sidebar"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="drawer"
        style={
          {
            "backgroundColor": "#c0c0c0",
            "position": "relative",
            "zIndex": -1,
          }
        }
      >
        <div
          className="navImage"
        >
          <img
            alt="Luther Rose"
            id="webjamwidelogo"
            src="https://upload.wikimedia.org/wikipedia/commons/thumb/a/ab/Lutherrose.svg/800px-Lutherrose.svg.png"
            style={
              {
                "marginLeft": 0,
                "marginRight": 0,
                "width": "86px",
              }
            }
          />
        </div>
        <div
          className="nav-list"
          style={
            {
              "width": "220px",
            }
          }
        >
          <p
            style={
              {
                "fontSize": "1px",
                "marginBottom": "2px",
              }
            }
          />
          <div
            className="menu-item"
            style={
              {
                "backgroundColor": "#244a8bff",
              }
            }
          >
            <p
              style={
                {
                  "color": "#fff",
                  "fontSize": "11pt",
                  "marginBottom": "2px",
                }
              }
            >
              <a
                className="menu-hover"
                href="http://bit.ly/CollegeLutheranDirections"
                style={
                  {
                    "color": "#88c1ff",
                  }
                }
              >
                <span>
                  210 S. College Ave
                </span>
              </a>
              <br />
              Salem, VA 24153
            </p>
          </div>
          <div
            className="menu-item"
            style={
              {
                "backgroundColor": "#244a8bff",
                "marginLeft": "0px",
              }
            }
          >
            <p
              style={
                {
                  "color": "#fff",
                  "fontSize": "11pt",
                  "marginBottom": "2px",
                }
              }
            >
              <span>
                ph: 
              </span>
              <a
                className="menu-hover"
                href="tel:5403894963"
                style={
                  {
                    "color": "#88c1ff",
                  }
                }
              >
                (540) 389-4963
              </a>
              <br />
              <span>
                fax: 
              </span>
              (540) 389-4980
              <br />
              <a
                href="mailto:office1@collegelutheran.org"
                style={
                  {
                    "color": "#88c1ff",
                    "wordWrap": "break-word",
                  }
                }
              >
                <span
                  className="menu-hover"
                >
                  office1@collegelutheran.org
                </span>
              </a>
            </p>
          </div>
          <div
            className="menu-item"
          >
            <a
              className="nav-link"
              href="/page2"
              onClick={[Function]}
            >
              <div
                style={
                  {
                    "display": "inline",
                  }
                }
              >
                <i
                  className="fas fa-users"
                />
                 
                <span
                  className="nav-item"
                >
                  page2
                </span>
              </div>
            </a>
          </div>
          <div
            className="menu-item"
          >
            <a
              className="nav-link"
              href="/"
              onClick={[Function]}
            >
              <div
                style={
                  {
                    "display": "inline",
                  }
                }
              >
                <i
                  className="fas fa-heart"
                />
                 
                <span
                  className="nav-item"
                >
                  Home
                </span>
              </div>
            </a>
          </div>
          <p
            style={
              {
                "fontSize": "6pt",
                "margin": 0,
                "padding": 0,
              }
            }
          >
             
          </p>
        </div>
      </div>
    </div>
    <div
      className="main-panel"
    >
      <span
        id="mobilemenutoggle"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
      >
        <i
          className="fas fa-bars"
        />
      </span>
      <div
        className="mainPanel"
      >
        <div
          className="swipe-area"
        />
        <div
          className="material-header home-header"
          id="header"
          style={
            {
              "backgroundColor": "#244a8bff",
            }
          }
        >
          <div
            className="headercontent"
          />
          <div>
            <div
              className="flex-header"
              style={
                {
                  "marginLeft": "15px",
                  "marginTop": "-18px",
                  "paddingLeft": "10px",
                  "paddingRight": "10px",
                }
              }
            >
              <h2
                className="header-text"
                style={
                  {
                    "fontSize": "25px",
                    "marginBottom": "-1px",
                    "marginTop": "1px",
                  }
                }
              >
                <a
                  className="header-text"
                  href="/"
                  style={
                    {
                      "textAlign": "left",
                      "textDecoration": "none",
                    }
                  }
                >
                  React Template
                </a>
              </h2>
              <p
                className="subTitle"
                style={
                  {
                    "color": "#f4c00eff",
                    "maxWidth": "100%",
                  }
                }
              >
                A React Template
              </p>
            </div>
          </div>
        </div>
        <div
          className="content-block"
          id="contentBlock"
          style={
            {
              "width": "auto",
            }
          }
        >
          <div>
            <h1>
              Page 1
            </h1>
            <div>
              <p>
                Some page one content here.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
